run_email_parse.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/python
+import re,sys
+import scrapy
+from scrapy.crawler import CrawlerProcess
+from scrapy.spiders import CrawlSpider, Rule
+from scrapy.linkextractors import LinkExtractor
+from scrapy import signals
+from scrapy.conf import settings
+from scrapy.xlib.pydispatch import dispatcher
+from selenium import webdriver
+
+settings.set('LOG_LEVEL', 'INFO')
+
+class EmailSpider(CrawlSpider):
+    name = 'email'
+
+      
+    def __init__(self, **kwargs):
+	self.allowd_domains = [kwargs['domain']]
+	self.start_urls= ["http://www."+kwargs['domain']]
+        self.rules = (
+            Rule(LinkExtractor(allow=(), allow_domains=(self.allowd_domains)), callback='parse_start_url', follow=True),
+	)
+        CrawlSpider.__init__(self) 
+	self.filename = kwargs['output']
+        self.driver = webdriver.Chrome()
+        dispatcher.connect(self.spider_closed, signals.spider_closed)
+	self.visited = {}
+	self.email_list = set()
+
+    def spider_closed(self, spider):
+	with open(self.filename,'w') as f:
+	    f.write('\n'.join(self.email_list))
+	    f.write('\n')
+        self.driver.quit() 
+    
+    def parse_start_url(self, response):
+	#extract ng-click web-page
+        additional_pages = response.css('span::attr(ng-click)').re(r'changeRoute\(\'(\w+)\'\)')
+        for url in additional_pages:
+	    if url not in self.visited:
+		self.visited[url] = 1
+                yield self.parse_jssite_url(response.url+'/'+url)
+	
+	#handle static dom
+	for sel in response.css('a[href*="'+self.allowd_domains[0]+'"]::attr(href)'):
+	    self.add_to_list(sel.extract())
+	
+    
+    def parse_jssite_url(self,url):
+	#handle dynamic dom
+        self.driver.get(url)
+        elms = self.driver.find_elements_by_partial_link_text('@jana.com')
+	for elm in elms:
+	    self.add_to_list(elm.get_attribute('href'))
+
+    def add_to_list(self, s):
+	m = re.match('(mailto\:)?([\w\.]+)@'+self.allowd_domains[0], s)
+	if m:
+            self.email_list.add('{0}@{1}'.format(m.groups()[1],self.allowd_domains[0]))
+
+
+if __name__ == "__main__":
+    if len(sys.argv) < 3:
+        print "You must enter domain name and output file name "
+	exit()
```

Comment:
You could potentially use a package like argparse (https://docs.python.org/2/library/argparse.html) to organize stuff like this.